packages/rocketchat-grant/server/index.js

```diff
@@ -12,9 +13,30 @@ import Providers, { middleware as providers } from './providers';
 import Settings from './settings';
 
 let grant;
+let storeAddress = process.env.SESSIONS_URL || 'mongodb://localhost:27017/rocketsession';
+
+if (storeAddress.indexOf('connectTimeoutMS=') < 0) {
```

Comment:
any special reason to enforce a `connectTimeoutMS`? do you think we should add the same to default Rocket.Chat Mongo connection?

Reply:
I would rather have my app crash fast and pm2 restart my processes right away, if database connection feels slow. Especially for sessions, and arguably to rocket main database as well.
Then again, I wouldn't want to be responsible for crashing existing deployments that may be working just fine enough.

Accounting for slow assets download, slow javascript processing, ... you want your backend to respond fast. On a distributed setup, having rocket workers fail based on high network latencies may help improve service responsiveness from the end-user point of view.
The default mongodb connection timeout is set to 30s. Being a connection timeout, it is not related to the queries you may send, the responses sizes, ... only on the time required for your TCP handshake to complete. As such, mongodb being at the core of rocket, we could want to lower that default to something that'ld make more sense in our context.

Then again, I may be wrong, I have my biases, ...